### PR TITLE
Remove debug output

### LINKED ISSUE #5: 'Expected errors' create output on stdout
image

But that is on a happy path when a repo is `init`ed that does not exist beforehand.

image

Maybe this is a debugging leftover?

In general the error handling here is not very sophistocated.
Only the caller of `repository_open()` can decide if the 'error' is an error or expected and will be handled.

We can leave this for now, but some other mechanism is needed for production.

### PR DESCRIPTION
**Pull request by Finii**
The wrapper functions output some error messages to stderr when something happens.
Sometimes the error is "expected" but the message is printed nonetheless.
For example
* A repository is to be opened or created...
* It tries to open the repo via libgit2
* If that fails a new repo is created

We have 2 possibilities...

1. Keep the error messages but only throw them when they are unexpected.
   For this we expand the LibGitWrapper to be a bit like a gul14::expected
   (which is, I just noticed, quite different from std::expected).
   This is implemented in the PR.
2. Just drop the error messages and hope all errors are handled by the wrapper users.

Fixes: #5 

I'm not sure what we want, 1 or 2 or something else. Please feel free to suggest something different, etc.
I know tests for the extension of LibGitWrapper are missing, I wanted to wait until I know we really want that ;-)

_Edit: Add last sentense_

### CODE REVIEW

**Comment by Finii** [comment]
I would be more comfortable with 2 if we hide the wrappers from our users; at the moment it is in a public header.
Then we could argue WE are experts and will never ignore errors and thus do not need the error messages at all.

```diff
--- a/include/libgit4cpp/meson.build
+++ b/include/libgit4cpp/meson.build
@@ -4,6 +4,9 @@ public_headers = [
     'GitRepository.h',
     'libgit4cpp.h',
     'LibGitPointer.h',
-    'wrapper_functions.h',
 ]
+
+private_headers = [
+    'wrapper_functions.h',
+]
```
;-)

File: include/libgit4cpp/wrapper_functions.h [hunk 1]
```diff
@@ -81,96 +82,94 @@ signature_default(git_repository* repo);
...
- * 
+ * Find the name of a branch on the remote.
+ * \param repo Pointer to repository object
+ * \param branch_name Local branch name, e.g. 'main', 'fix-bugs'
+ * \return name on remote , e.g. 'origin/main' or 'origin/fix-bugs'
 */
 std::string
 branch_remote_name(git_repository* repo, const std::string& branch_name);
 
-/** \}*/
```
Comment (alt-graph) [inline include/libgit4cpp/wrapper_functions.h:172]: I believe this comment closes the group opened above with `\defgroup`?

**Review by alt-graph — COMMENTED** [review]
Interesting idea. Let's discuss this in the meeting.

File: src/GitRepository.cc [hunk 2]
```diff
@@ -103,12 +103,12 @@ void GitRepository::init(const std::filesystem::path& file_path)
 {
     repo_ = repository_open(repo_path_);
 
-    if (repo_.get() == nullptr)
+    if (not repo_)
```
Comment (wyrnat) [inline src/GitRepository.cc:106]: > if (git_repository_open(&repo, repo_path.c_str()))
>         return gul14::cat("repository_open: ", git_error_last()->message);

If repository_open fails, it returns a string with error message. How can "not repo_" be true if the string is never empty?

Comment (wyrnat) [reply src/GitRepository.cc:106]: Nevermind, I saw you overloaded the LiGitPointer constructor with a string. Looks good!

**Review by wyrnat — APPROVED** [review]
:+1:

**Comment by Finii** [comment]
> Interesting idea. Let's discuss this in the meeting.

Sorry I am at the MTCA workshop 😒

File: src/GitRepository.cc (hunk 2 above)
Comment (Finii) [reply src/GitRepository.cc:106]: > How can "not repo_" be true

Well, as with gul14::expected and similar types std::expected or std::optional, see below.
I guess this is what ppl expect nowadays ;-)

Screenshot 2023-12-06 at 11 13 30

Ah, I see I forgot the `explicit` :-/

Comment (Finii) [reply src/GitRepository.cc:106]: Fixed explicit

File: include/libgit4cpp/wrapper_functions.h (hunk 1 above)
Comment (Finii) [reply include/libgit4cpp/wrapper_functions.h:172]: Ah yes, never heard about such groups.

Reintroduced the closing curly, but - as suggested by the Doxygen docs - put an additional comment there that this closes the group:

Screenshot 2023-12-06 at 17 51 07

**Comment by Finii** [comment]
Did you talk about this in the meeting today?
Shall I do something or wait till next week's meeting?

Fixed the two issues in the original commits, force push.

**Comment by Finii** [comment]
> Did you talk about this in the meeting today?

Obviously some discussion is needed for the change. That is totally fine.
But to get `taskolib` ready we need some solution for the debug messages on `stdout`.

So I created a new PR:
* #14 

That PR has all this PR has, just that it does not store/throw the messages.
All other code improvements went over to the new PR.
And it additionally renamed `GitLibPointer` which is imho a misnomer, please see reasoning over there.

If we want the storing/throwing of the messages I can/will add that to #14 or a further new PR after a release or whatever.

**Pull request closed without merging**